### ISSUE TITLE
TAB completion in python 3.8

### ISSUE THREAD

**Issue by kokostek** [ipython/ipython#12496]
It seems to be broken in my conda env with python 3.8 when I try to tab-complete imported module.

Steps to reproduce:

1. Define a simple module:
```
# foo.py
def bar():
    pass
```

2. Create conda environment with python=3.8 and ipython=7.16.1 (the latest one at the moment available in conda repo for py 3.8).
```
conda create -n py38 python=3.8 ipython=7.16.1
conda activate py38
```

3. Run ipython from new env:
```
ipython
```

4. Import test module and try to do tab completion:
```
In [1]: import foo
In [2]: foo.<TAB>
```

5. Get only the file name suggestion (`foo.py`) but no `foo.bar()`.

With python 3.7 completion seems to work fine, i.e.:
```
conda create -n py37 python=3.7 ipython=7.16.1
conda activate py37
ipython
In [1]: import foo
In [2]: foo.<TAB>
```
results in several suggestions including `foo.bar()`.

**Comment by kokostek** [comment]
Here is my `conda list` output from both envs:
```
# packages in environment at C:\Users\User\Miniconda3\envs\py37:
#
# Name                    Version                   Build  Channel
backcall                  0.2.0                      py_0  
ca-certificates           2020.6.24                     0  
certifi                   2020.6.20                py37_0  
colorama                  0.4.3                      py_0  
decorator                 4.4.2                      py_0  
ipython                   7.16.1           py37h5ca1d4c_0  
ipython_genutils          0.2.0                    py37_0  
jedi                      0.17.2                   py37_0  
openssl                   1.1.1g               he774522_1  
parso                     0.7.0                      py_0  
pickleshare               0.7.5                 py37_1001  
pip                       20.2.2                   py37_0  
prompt-toolkit            3.0.5                      py_0  
pygments                  2.6.1                      py_0  
python                    3.7.7                h81c818b_4  
setuptools                49.6.0                   py37_0  
six                       1.15.0                     py_0  
sqlite                    3.32.3               h2a8f88b_0  
traitlets                 4.3.3                    py37_0  
vc                        14.1                 h0510ff6_4  
vs2015_runtime            14.16.27012          hf0eaf9b_3  
wcwidth                   0.2.5                      py_0  
wheel                     0.34.2                   py37_0  
wincertstore              0.2                      py37_0  
zlib                      1.2.11               h62dcd97_4  
```
```
# packages in environment at C:\Users\User\Miniconda3\envs\py38:
#
# Name                    Version                   Build  Channel
backcall                  0.2.0                      py_0  
ca-certificates           2020.6.24                     0  
certifi                   2020.6.20                py38_0  
colorama                  0.4.3                      py_0  
decorator                 4.4.2                      py_0  
ipython                   7.16.1           py38h5ca1d4c_0  
ipython_genutils          0.2.0                    py38_0  
jedi                      0.17.0                   py38_0  
openssl                   1.1.1g               he774522_1  
parso                     0.8.0                      py_0  
pickleshare               0.7.5                 py38_1000  
pip                       20.2.2                   py38_0  
prompt-toolkit            3.0.5                      py_0  
pygments                  2.6.1                      py_0  
python                    3.8.5                he1778fa_0  
setuptools                49.6.0                   py38_0  
six                       1.15.0                     py_0  
sqlite                    3.32.3               h2a8f88b_0  
traitlets                 4.3.3                    py38_0  
vc                        14.1                 h0510ff6_4  
vs2015_runtime            14.16.27012          hf0eaf9b_3  
wcwidth                   0.2.5                      py_0  
wheel                     0.34.2                   py38_0  
wincertstore              0.2                      py38_0  
zlib                      1.2.11               h62dcd97_4  
```

**Comment by Carreau** [comment]
1) Update jedi to 0.17.2, tab completion is powered by jedi
2) can you try IPython 7.17 ?

**Comment by kokostek** [comment]
Problem resolved with jedi 0.17.2. Upgrading ipython to 7.17.0 makes no difference.
I think we can close this.
